src/lib/disk_cache/filesystem/disk_cache.ml

```diff
@@ -10,36 +10,106 @@ module Make (B : sig
   include Binable.S
 end) =
 struct
```

Comment:
I may consider removing FS disk cache as a whole: 
- We're not using it
- It performs worse than LMDB
- It's a burden to implement similar logic twice here